[UI] New volumes page

alexis-ld opened the pull request:
**Component**:

ui, volumes

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

We want to adapt the volumes page so it looks like the new node page (tab based).

**Summary**:

I refactored most of the Volume view. There were also small adaptations compared to the base view (alerts filtering in URL, ...) as described in #2751 .

Some components (ActiveAlertsFilters, ActiveAlertsCounter) were designed to be re-usable in other views.

EDIT : Missing tabs (metrics and details) were added in #2866 targeting this branch.

![Screenshot 2020-10-15 at 15 41 13](https://user-images.githubusercontent.com/3480526/96139475-35faa480-0eff-11eb-8a0f-189b2a388aaa.png)
![Screenshot 2020-10-15 at 15 41 26](https://user-images.githubusercontent.com/3480526/96139487-372bd180-0eff-11eb-8634-ff0daacdf242.png)
![Screenshot 2020-10-15 at 15 41 46](https://user-images.githubusercontent.com/3480526/96139497-38f59500-0eff-11eb-8d25-c2d5f245d039.png)
![Screenshot 2020-10-15 at 15 41 59](https://user-images.githubusercontent.com/3480526/96139502-3a26c200-0eff-11eb-94df-965911f9351a.png)



**Acceptance criteria**: 

* Volumes page right panel is separated in four tabs Overview, Alerts, Metrics, Details
* Total number of alerts is displayed in Alerts tab header
* It is possible to filter by alert severity on the Alert tab
* Clicking on an alert category in the overview tab opens the alerts tab and filters by the corresponding category
* When switching to another volume the selected tab is preserved
---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2751

bert-e commented:
# Hello alexis-ld,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2751-new-volumes-page` into
`development/2.6`
* [w/2.7/improvement/2751-new-volumes-page](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/2751-new-volumes-page) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal

bert-e commented:
# History mismatch



Merge commit #af8746b38abb5dd7fcb8fd93df99d6f8661084a1 on the integration branch
`w/2.7/improvement/2751-new-volumes-page` is merging a branch which is neither the current
branch `improvement/2751-new-volumes-page` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/2751-new-volumes-page` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexis-ld commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2751-new-volumes-page` into
`development/2.6`
* [w/2.7/improvement/2751-new-volumes-page](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/2751-new-volumes-page) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal

ChengYanJin commented on ui/src/components/ActiveAlertsCounter.js:
```diff
@@ -0,0 +1,102 @@
+} from '@scality/core-ui/dist/style/theme';
+
+import {
+    STATUS_WARNING,
```
Can we please make sure we have the same prettier setting? So that our codes have the same format. 

```javascript
{
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "all"
}
```

gdemonet replied:
basically use the file at `ui/.prettierrc`

ChengYanJin commented on ui/src/components/VolumeDetailCard.js:
```diff
@@ -195,6 +211,8 @@ const VolumeDetailCard = (props) => {
     <VolumeDetailCardContainer>
       <VolumeInformation>
         <VolumeNameTitle data-cy="volume_detail_card_name">
+          <CircleStatus className="fas fa-circle" status={health} />
```
```suggestion
          <CircleStatus className="fas fa-circle fa-2x" status={health} />
```
According to the design, please use the bigger size of the icon.

ChengYanJin commented on ui/src/components/VolumeDetailCard.js:
```diff
@@ -195,6 +211,8 @@ const VolumeDetailCard = (props) => {
         <VolumeNameTitle data-cy="volume_detail_card_name">
+          <CircleStatus className="fas fa-circle" status={health} />
+          &nbsp;
           {name}
```
the font size of the `Volume Name Title` should be ${fontSize.larger} according to the design

ChengYanJin commented on ui/src/components/ActiveAlertsFilters.js:
```diff
@@ -0,0 +1,92 @@
+
+    return (
+        <Dropdown
+        items={items}
```
The items here should be all the available timespan **exclude** the current selection. So we need to filter the metricsTimeSpanItems and pass them to items.

ChengYanJin commented on ui/src/components/VolumeActiveAlertsCard.js:
```diff
@@ -166,8 +182,17 @@ const ActiveAlertsCard = (props) => {
+        <div>
+          {intl.translate('active_alerts')}
+          &nbsp;&nbsp;
+          <TextBadge>{alertlist?.length}</TextBadge>
```
After discussing with our UX, here we could remove the `<TextBadge>` since we have the same thing in the Alerts tab title. No need to duplicate. 

Same for the TextBadge in the Alert tab.

Thanks.

ChengYanJin commented on ui/src/components/ActiveAlertsFilters.js:
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Dropdown } from '@scality/core-ui';
+
```
Please remove the empty line here, same for the rest. Thanks!

ChengYanJin commented on ui/src/components/ActiveAlertsFilters.js:
```diff
@@ -0,0 +1,92 @@
+    const { baseLink } = props;
+    const history = useHistory();
+    const location = useLocation();
+    const query = new URLSearchParams(location.search);
```
FYI, we have `useQuery` hook. Would you think it's more efficient to `useLocation()` directly than `useHistory().location`?

```javascript
// A custom hook that builds on useLocation to parse the query string.
export const useQuery = () => {
  return new URLSearchParams(useLocation().search);
};
```

alexis-ld replied:
No, you are absolutely right, replacing with the hook in this component, thanks :) .

In ActiveAlertsCounter component though, I should likely keep the location item. I need independent URLSearchParams for link generation and thus I am creating those in functions. Rules of hooks prevent the use of useQuery() in a function :).
But definitely in ActiveAlertsFilers I am replacing by the hook. :)

ChengYanJin commented on ui/src/components/VolumeDetailCard.js:
```diff
@@ -117,6 +124,12 @@ const LabelValue = styled.span`
+`;
+
+
 const VolumeDetailCard = (props) => {
```
Not directly related to this change, but can we prevent the text of  `Used by` and `Labels` from wrapping? since we have enough space after the Volume Usage Progress Bar.

alexis-ld replied:
Not sure... the values are still limited by the column splitting the panel in half. If we want to do that we should refactor the layout so the left column is spit in two part (one that is at Usage progress bar level, and one that is below), that can be tricky.
Do you want me to try that ? The wrapping is not so bad on wide screens.

ChengYanJin replied:
From UX point of view, we would avoid to have "Overflow content" if there is enough space.
So, yeah, please refactor the layout.

ChengYanJin commented on ui/src/components/VolumeActiveAlertsCard.js:
```diff
@@ -67,16 +76,22 @@ const NoActiveAlerts = styled.div`
 
 const ActiveAlertsCard = (props) => {
-  const { alertlist, PVCName } = props;
+  const { alertlist, PVCName, name } = props;
+  const match = useRouteMatch();
```
In order to get the selected volumes, we can rely on `react-router's` Route matching. So no need to pass `name` to component `<ActiveAlertsCard>`

We forgot to declare the parameter to VolumePage In Route, in this way would be easier to retrieve the name

Layout.js    
```javascript
<PrivateRoute path="/volumes/:name" component={VolumePage} />
```

alexis-ld replied:
Indeed :) .

Ended up with : 

```<PrivateRoute path="/volumes/:name?" component={VolumePage} />``` 

Without the ? the VolumePage component would not render if no name was specified in the URL.

ChengYanJin commented on ui/src/components/VolumeActiveAlertsCard.js:
```diff
@@ -166,8 +182,17 @@ const ActiveAlertsCard = (props) => {
+      <ActiveAlertsTitle>
+        <div>
+          {intl.translate('active_alerts')}
+          &nbsp;&nbsp;
```
Would be better to use the padding to be more accurate.

ChengYanJin commented on ui/src/components/VolumeListTable.js:
```diff
@@ -400,12 +401,13 @@ const VolumeListTable = (props) => {
     const query = new URLSearchParams(location.search);
     const isAddNodeFilter = query.has('node');
+    const parsedUrl = location.pathname.split('/');
 
     // there are two possiable URLs
     if (isAddNodeFilter) {
       history.push(`/volumes/${row.values.name}?node=${nodeName}`);
     } else {
-      history.push(`/volumes/${row.values.name}`);
+      history.push(`/volumes/${row.values.name}${parsedUrl.length === 4 ? `/${parsedUrl[3]}` : ''}${query.toString() && `?${query.toString()}`}`);
     }
   };
```
I would suggest not to parse the URL.

```javascript
    if (isTabSelected) {
      const newPath = location.pathname.replace(
        /\/volumes\/[^/]*\//,
        `/volumes/${volumeName}/`,
      );
      history.push({
        pathname: newPath,
        search: query.toString(),
      });
    } else {
      history.push({
        pathname: `/volumes/${volumeName}/overview`,
        search: query.toString(),
      });
    }
```

alexis-ld replied:
Much cleaner indeed :) thanks

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -1,40 +1,72 @@
+  const theme = useSelector((state) => state.config.theme);
+
   const currentVolumeName =
     history?.location?.pathname?.split('/').slice(2)[0] || '';
```
Could retrieve the currentVolumeName by `react-router's` Route Matching.

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -131,39 +182,53 @@ const VolumePageContent = (props) => {
+          <Tabs activeColor={theme.brand.primary} items={tabsItems}>
+            {
+              isOverviewPage &&
+              <VolumeDetailCard
```
We may want to rename this component to `VolumeOverviewTab`

alexis-ld replied:
Also renamed VolumeAlertsTab. :)

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -68,59 +100,78 @@ const VolumePageContent = (props) => {
-    volumeIOPSWrite,
-    queryStartingTime,
-  };
+  // const deviceName = volume?.status?.deviceName;
+  // let instance;
+
+  // if (!node.internalIP) {
+  //   // find the node name of this volume
+  //   const nodeName = volume?.spec?.nodeName;
+  //   const currentNode = nodes.find((node) => node.name === nodeName);
+  //   instance = `${currentNode?.internalIP}:${PORT_NODE_EXPORTER}`;
+  // } else {
+  //   instance = `${node?.internalIP}:${PORT_NODE_EXPORTER}`;
+  // }
+
+  // const queryStartingTime = volumeStats?.queryStartingTime;
+  // const volumeUsage = volumeStats?.volumeUsage?.find(
+  //   (vU) => vU.metric.persistentvolumeclaim === PVCName,
```
Please keep those codes from comments.

alexis-ld replied:
The reason I commented this is because all this code will be re-enabled when I will add the metrics tab. I wanted to prevent this from being reviewed as "new code" since it was already there and it is a big chunk of code.

Should I delete it and re-add it later then ?

ChengYanJin replied:
IMO it makes sense to keep the code, even though we don't directly use them in this PR. 
It brings a lot of noise by commenting on it or removing it.

alexis-ld replied:
I agree :) . The issue was that eslint was making the whole build fail for unused variables if I left the code in place without commenting it.

ChengYanJin replied:
Indeed. we'll have ESLint error for the unused vars.
I would suggest you to locally disable the Rules with Inline Comments. And add a little comment explaining it's a temporary disable until refactoring is done ;)

```javascript
  // Temporarily disable the `no-unused-vars` lint until the refactoring is complete.
  // eslint-disable-next-line no-unused-vars
  const volumeMetricGraphData = {
    volumeUsage,
    volumeThroughputWrite,
    volumeThroughputRead,
    volumeLatencyWrite,
    volumeLatencyRead,
    volumeIOPSRead,
    volumeIOPSWrite,
    queryStartingTime,
  };
```

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -131,39 +182,53 @@ const VolumePageContent = (props) => {
+              />
+              }
+          </Tabs>
+          {/* <ActiveAlertsCard
```
Please remove the code it's not used anymore.

ChengYanJin commented on ui/src/components/VolumeDetailCard.js:
```diff
@@ -281,9 +299,23 @@ const VolumeDetailCard = (props) => {
         {condition === VOLUME_CONDITION_LINK && (
           <VolumeUsage>
-            <VolumeUsageTitle>Volume Usage</VolumeUsageTitle>
+            <VolumeSectionTitle>{intl.translate('volume_usage')}</VolumeSectionTitle>
```
The title should be just `Usage` according to the design.
And thanks for the translations.

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -131,39 +182,53 @@ const VolumePageContent = (props) => {
-            pVList={pVList}
-          ></VolumeDetailCard>
-          <ActiveAlertsCard
+            {
```
We could have Switch inside the `<Tab>`, would you think it's easier to read?

```javascript
<Switch>
  <Route
    path={`/volumes/${volumeName}/overview`}
    component={VolumeOverviewTab}
   />
  ...
</Switch>
```

alexis-ld replied:
Honestly, I prefer the 
`isOverviewPage &&`
approach.

We have to define the isXXXXPage variables anyway to tell the Tabs component which one is selected so I think we should as well use these variables to decide what to render since these are already defined.
It avoid to create more wrappers with the Route Switch and also IMO is easier to read.

But I totally understand if you prefer the Switch and want me to follow the existing coding-style. Let me know if I should change ? :)

ChengYanJin replied:
Yes. So let's follow the current coding-style ;)

ChengYanJin reviewed (changes requested):
Thanks for the work! 
Here are a few things I would like to mention before the review: 

- Please make sure we have the same prettier setting.
- Please splits into commits with proper message. [Commit Best Practices](https://github.com/scality/metalk8s/blob/development/2.6/docs/developer/development/commit.rst). It will be easier for us to review the code.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

Commit:
ui/volumes: Add translations for new volume page
Refs: #2751

Commit:
ui/volumes:
Add optional parameter to /volumes/ route to use volume name more easily in sub components
Refs: #2751

Commit:
ui/volumes:
Re-usable components to filter/count alerts for a volume
Refs: #2751

Commit:
ui/volumes:
TextBadge component to display a counter in a block

Commit:
ui/volumes:
Refactor VolumeAlertsCard component to serve as a tab in new Volumes view
Refs: #2751

Commit:
ui/volumes:
Refactors VolumeDetailsCard as the overview tab for the new volumes page
Refs: #2751

Commit:
ui/volumes:
Adapts Volume table to forward URLs properly to keep tabs selection on the right side.
Refs: #2751

Commit:
ui/volumes:
Add tab-based right panel in on Volumes page
Refs: #2751

bert-e commented:
# History mismatch



Merge commit #82a3b5f03f3624f5a558a0a7c2486b0423e7a3b8 on the integration branch
`w/2.7/improvement/2751-new-volumes-page` is merging a branch which is neither the current
branch `improvement/2751-new-volumes-page` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/2751-new-volumes-page` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexis-ld commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2751-new-volumes-page` into
`development/2.6`
* [w/2.7/improvement/2751-new-volumes-page](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/2751-new-volumes-page) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

Commit:
ui/volumes:
Use Switch from react-router-dom in tabs to display selected content to match coding style.

Refs: #2751

bert-e commented:
# History mismatch



Merge commit #5700c91d13a491b1b74ec925099c44ac7e05d429 on the integration branch
`w/2.7/improvement/2751-new-volumes-page` is merging a branch which is neither the current
branch `improvement/2751-new-volumes-page` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/2751-new-volumes-page` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexis-ld commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2751-new-volumes-page` into
`development/2.6`
* [w/2.7/improvement/2751-new-volumes-page](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/2751-new-volumes-page) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

ChengYanJin committed:
ui/theme: Update the them for light/dark mode

- Update the dark mode with final version
- Update the status color of light mode with proposal, but keep the primary/text/border as the previous version,
  since it doesn't work well.

Commit:
ui/volumes:
Adapt colors based on the new theme variables to fit to the design

Refs: #2751

Commit:
ui/volumes:
Make node name clickable in overview tab. On click it redirects to the corresponding node page.

Refs: #2751

ChengYanJin commented on ui/src/components/ActiveAlertsCounter.js:
```diff
@@ -0,0 +1,100 @@
+  ).length;
+  const warningCounter = alerts?.filter(
+    (item) => item?.labels?.severity === STATUS_WARNING,
+  ).length;
```
Hi @alexis-ld,
Thanks for extracting those 2 components so we can share and reuse them. 

If I were you, I would have passed the number of `criticalCounter` and `warningCounter` to `ActiveAlertsCounter` instead of `alerts`.  Since what we really need is only the number ;)

ChengYanJin commented on ui/src/components/ActiveAlertsFilters.js:
```diff
@@ -0,0 +1,98 @@
+`;
+
+const ActiveAlertsFilter = (props) => {
+  const { baseLink } = props;
```
I think it makes sense to remove the `baseLink` props since the URL should only change the query string in this component.

ChengYanJin commented on ui/src/components/CommonLayoutStyle.js:
```diff
@@ -64,3 +64,12 @@ export const TabTitle = styled.div`
+  border-radius: 4px;
+  font-size: ${fontSize.small}
+  font-weight: ${fontWeight.bold}
+`;
```
I would prefer to add  `margin-left: ${padding.smaller};` for TextBadge instead of `&nbsp;&nbsp;` in the tab title.

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -131,49 +184,57 @@ const VolumePageContent = (props) => {
-            volumeCondition={currentVolume.status}
-            // Hardcode the port number for prometheus metrics
-          ></MetricGraphCard>
+          <Tabs activeColor={theme.brand.primary} items={tabsItems}>
```
For the `<Tabs>` we get the feedback that the contrast of the unselected tab is not good, they are not very visible.

Please update with the new design: 
https://www.sketch.com/s/f7ae733f-222f-4335-b742-a86d25dda872/a/n1OpOz

Note that the Sketch is node page, but we want to apply to volume page as well.
The height of the tab title should be 40px according to the design.

Thanks!

ChengYanJin commented on ui/src/components/ActiveAlertsCounter.js:
```diff
@@ -0,0 +1,100 @@
+
+    switch (props.status) {
+      case STATUS_WARNING:
+        color = theme.warning;
```
```suggestion
        color = theme.alert;
```

ChengYanJin commented on ui/src/containers/VolumePageContent.js:
```diff
@@ -119,8 +154,26 @@ const VolumePageContent = (props) => {
   };
 
+  const isAlertsPage = location.pathname.endsWith('/alerts');
+  const isOverviewPage = location.pathname.endsWith('/overview') || !isAlertsPage;
```
```suggestion
  const isOverviewPage = location.pathname.endsWith('/overview');
```

ChengYanJin commented on ui/src/components/VolumeOverviewTab.js:
```diff
@@ -195,6 +210,8 @@ const VolumeDetailCard = (props) => {
     <VolumeDetailCardContainer>
       <VolumeInformation>
         <VolumeNameTitle data-cy="volume_detail_card_name">
+          <CircleStatus className="fas fa-circle fa-2x" status={health} />
```
Ha. If you use the component `<CircleStatus/>` the double size may not work.

You may want to cherry-pick [this commit](https://github.com/scality/metalk8s/pull/2868/commits/778b284d107e87614d1568af1ebd3b6225793f1a) (once it's merged) and properly use this component.

alexis-ld replied:
I tried with your commit but it seems the CIRCLE_SIZE_... constants are not included in the commit. I removed the 2x for now it seems it is still displaying properly :) . Will update when your commit will be merged.

ChengYanJin commented on ui/src/components/VolumeOverviewTab.js:
```diff
@@ -195,6 +210,8 @@ const VolumeDetailCard = (props) => {
       <VolumeInformation>
         <VolumeNameTitle data-cy="volume_detail_card_name">
+          <CircleStatus className="fas fa-circle fa-2x" status={health} />
+          &nbsp;
```
We may want to use padding to add space.

ChengYanJin commented on ui/src/components/VolumeOverviewTab.js:
```diff
@@ -195,6 +210,8 @@ const VolumeDetailCard = (props) => {
+          &nbsp;
           {name}
         </VolumeNameTitle>
         <InformationSpan>
```
You may want to reflector the layout to avoid the label from wrapping as we discussed last time?

alexis-ld replied:
I did that update in #2866 to avoid conflicting with the scrollable tab commit I did on the other PR.

ChengYanJin reviewed (changes requested):
Hello @alexis-ld ,

Thanks for the work!
Some comments about the overview and alerts tabs. I'll give my review of metrics and details review in another PR.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/2751-new-volumes-page` and the
destination branch `development/2.6`.

Please resolve the conflict on **the feature branch** (`improvement/2751-new-volumes-page`).

```bash
 $ git fetch
 $ git checkout origin/improvement/2751-new-volumes-page
 $ git merge origin/development/2.6
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/2751-new-volumes-page
```

ChengYanJin commented on ui/src/components/ActiveAlertsCounter.js:
```diff
@@ -0,0 +1,100 @@
+    if (existing.indexOf(status) === -1) {
+      query.set('severity', status);
+    }
+    return `${baseLink}?${query.toString()}`;
```
IMO, we could remove the baseLink in component `<ActiveAlertCounter/>` as well
What we need is too replace the `/overview` by `/alerts` what don't really need to know the pathname before.
```javascript
url = url.replace(/\/overview/, '/alerts');
```

```suggestion
    return `${url}?${query.toString()}`;
```

Commit:
ui/volumes:
Refactor ActiveAlertsCounter component to pass counters directly from props.

Commit:
ui/volumes:
Move URL handling directly into ActiveAlertsFilter component.

Commit:
ui/volumes: Remove &nbsp's and use margin instead.

Commit:
ui/volumes:
Update color code in ActiveAlertsCounter to fit design.

Commit:
ui/volumes:
Remove condition for overview tab selection as /overview is now default in the URL.

Refs: #2751

Commit:
ui/volumes:
Remove &nbsp; and use padding instead.

Commit:
ui/volumes: Removed fa-2x property.

Commit:
ui/volumes:
Remove baselink prop from ActiveAlertsCounter component handling the URL in the component directly.

Commit:
ui/volumes:
Changed tab styling to match the new design. Improved contrast and readability.

Refs: #2751

Commit:
Merge remote-tracking branch 'origin/development/2.6' into improvement/2751-new-volumes-page

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.7/improvement/2751-new-volumes-page` with contents from `improvement/2751-new-volumes-page`
and `development/2.7`.

Please resolve the conflict on **the integration branch** (`w/2.7/improvement/2751-new-volumes-page`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.7/improvement/2751-new-volumes-page
 $ git pull  # or "git reset --hard origin/w/2.7/improvement/2751-new-volumes-page"
 $ git merge origin/development/2.7
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/2751-new-volumes-page
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.7/improvement/2751-new-volumes-page
```

Commit:
ui/volumes:
Fix missing import after rebase in CommonLayoutStyle.js

alexis-ld commented:
@ChengYanJin I made all the changes, rebased from development 2.6 and fixed the conflicts :) .
Will fix the integration conflicts once we will have merged the missing pieces from #2866 to fix all conflicts at once.

ChengYanJin commented:
merged in PR #2866.
